### PR TITLE
Added check for mtime to be integer

### PR DESCRIPTION
**Pull request by virajparimi**
## Description

Added check in File.php of dav app in order to ensure that `$request->server['HTTP_X_OC_MTIME']` is an integer. If it is not then throws a `BadRequest` exception.

## Related Issue

[#27437](https://github.com/owncloud/core/issues/27437)

## How Has This Been Tested?

Have not tested it thoroughly. 
Chrome and Ubuntu 14.04

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/dav/lib/Connector/Sabre/File.php [hunk 1]
```diff
@@ -206,8 +206,13 @@ public function put($data) {
...
 			if (isset($request->server['HTTP_X_OC_MTIME'])) {
-				if ($this->fileView->touch($this->path, $request->server['HTTP_X_OC_MTIME'])) {
-					header('X-OC-MTime: accepted');
+				if( is_int($request->server['HTTP_X_OC_MTIME']) ) {
+					if ($this->fileView->touch($this->path, $request->server['HTTP_X_OC_MTIME'])) {
+						header('X-OC-MTime: accepted');
+					}
+				}
+				else {
+					throw new BadRequest('expected mtime to be integer');
```
Comment (PVince81) [inline apps/dav/lib/Connector/Sabre/File.php]: => "X-OC-MTime header must be an integer" would be a bit clearer

**Comment by PVince81** [comment]
Code looks good otherwise, thanks !

**Comment by viraj96** [comment]
@PVince81 done

**Comment by PVince81** [comment]
Please test with the desktop client, apparently the value is not recognized as int by `is_int`, you might need to use another approach to detect this.

``'
// => eval: $request->server['HTTP_X_OC_MTIME']
$evalResult                      = (string[10]) '1491982450';

// => eval: is_int($request->server['HTTP_X_OC_MTIME'])
$evalResult                      = (bool) '0';
```

**Comment by viraj96** [comment]
Ahh, so the $request->server['HTTP_X_OC_MTIME'] is a string and not an integer. Okay, one possible solution is to use `strpos` to get the position of `.` if any, otherwise it would return `false` and then everything else remains same. What do you think?

**Comment by PVince81** [comment]
`strpos` is not acceptable because it wouldn't catch other cases where someone sends an invalid value (for example if someone is crazy enough to send a ISO formatted date instead of an int)

**Comment by individual-it** [comment]
the is_int approach has also the problem that negative values would be accepted
I've changed the File.php so that put() should be easier testable and added some tests. 
@viraj96 I've made a PR to your fork https://github.com/viraj96/core/pull/1
You can think of more tests

**Comment by PVince81** [comment]
any update ?

**Comment by viraj96** [comment]
@PVince81 Hi, I have been busy with my End Semester Exams. Will work on it once they get over by this weekend. Sorry for the delay caused.

**Comment by mrow4a** [comment]
Any update here, or can I review?

**Comment by individual-it** [comment]
here is an alternative approach #28066  don't throw any errors but simply make sure the value is legal

**Comment by mrow4a** [comment]
This one can be closed since the other approach has been merged right?

**Pull request closed without merging**